crates/astria-composer/src/searcher/bundler.rs

```diff
@@ -0,0 +1,133 @@
+use std::{
+    collections::VecDeque,
+    future::{
+        self,
+        Ready,
+    },
+    mem,
+};
+
+use astria_core::sequencer::v1alpha1::{
+    transaction::{
+        action::SequenceAction,
+        Action,
+    },
+    ROLLUP_ID_LEN,
+};
+use tracing::debug;
+
+#[derive(Debug, thiserror::Error)]
+enum SizedBundleError {
+    #[error("bundle does not have enough space left for the given sequence action")]
+    NotEnoughSpace(SequenceAction),
+}
+
+/// A bundle sequence actions to be submitted to the sequencer. Maintains the total size of the
+/// bytes pushed to it and enforces a max size in bytes passed in the constructor. If an incoming
+/// `seq_action` won't fit in the buffer it is flushed and a new bundle is started.
+pub(super) struct SizedBundle {
```

Comment:
This `SizedBundle` should probably be an implementation detail of this module and not be leaked outside of it.